tests/Metadata/MetaDataBaseTest.php

```diff
@@ -228,9 +228,27 @@ public function testMissingField(string $expectedField, string $exception = null
      */
     public function testOptionalFields(string $optionalField, $value): void
     {
+        $optionalField = explode(':', $optionalField);
+
         $metadata = json_decode($this->clientStorage[$this->validJson], true);
-        static::nestedChange(explode(':', $optionalField), $metadata, $value);
-        $json = json_encode($metadata);
+        static::nestedChange($optionalField, $metadata, $value);
+        $this->assertDataIsValid($metadata);
```

Comment:
Moved the assertion to its own method, since we need to use it more than once now.